crates/flox-rust-sdk/src/models/search.rs

```diff
@@ -0,0 +1,261 @@
+        let mut results = Vec::new();
+        for maybe_line in bytes.lines() {
+            let text = maybe_line?;
+            match serde_json::from_str(&text) {
+                Ok(search_result) => results.push(search_result),
+                Err(_) => {
+                    // TODO: Errors are currently emitted to stdout as JSON, but there's no spec for the errors.
+                    //       For now if we can't turn the text into a SearchResult, we assume that it's an
+                    //       error message. If parsing that into a serde_json::Value fails, something else went
+                    //       pretty wrong.
+                    //
+                    //       Once there's a spec for the error messages we can parse this into a typed container.
+                    let err = Value::from_str(&text)?;
+                    return Err(SearchError::PkgDb(err));
+                },
+            };
+        }
+        Ok(SearchResults { results })
+    }
+}
+
+/// A package search result
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SearchResult {
+    /// Which input the package came from
+    pub input: String,
+    /// The attribute path of the package inside the input
+    #[serde(rename = "path")]
+    pub attr_path: Vec<String>,
+    /// The package name
+    pub pname: Option<String>,
+    /// The package version
+    pub version: Option<String>,
+    /// The package description
+    pub description: Option<String>,
+    /// Whether the package is marked "broken"
+    pub broken: Option<bool>,
+    /// Whether the package has an unfree license
+    pub unfree: Option<bool>,
+    /// Which license the package is licensed under
+    pub license: Option<String>,
+}
+
+#[cfg(test)]
+mod test {
+    use std::process::{Command, Output};
+    use std::str::FromStr;
+
+    use anyhow::Error;
+
+    use super::*;
+    const PKGDB: &'_ str = env!("PKGDB_BIN");
+
+    fn call_pkgdb(params: &SearchParams) -> Result<Output, Error> {
+        let params_json = serde_json::to_string(params).unwrap();
+        // Useful for debugging
+        // eprintln!("json input:\n{}", params_json);
+        let output = Command::new(PKGDB)
+            .arg("search")
+            .arg("--quiet")
+            .arg(params_json)
+            .output();
+        output.map_err(Error::from)
+    }
+
+    fn assert_no_err_msg(stderr: Vec<u8>) {
+        if !stderr.is_empty() {
+            let err_msg = String::from_utf8(stderr).unwrap();
+            // We know this will fail, but this way we'll get to see the error message from pkgdb
+            assert_eq!(String::from(""), err_msg);
+        }
+    }
+
+    #[test]
+    fn serializes_search_params() {
+        let params = SearchParams {
+            query: Query::from_str("hello@2.12.1").unwrap(),
+            ..SearchParams::default()
+        };
+        let results = call_pkgdb(&params).unwrap();
+        assert_no_err_msg(results.stderr);
```

Comment:
Should there be some more assertions about results here? Same for the deserializes test. I was wondering if you aren't doing that because you're trying to keep the test more focused? But I guess it's already an integration test since you're actually calling `pkgdb`...

Reply:
This is intentionally minimal for exactly that reason. We have integration tests, and they did their job when we discovered the catalog issue.

Reply:
Fixed